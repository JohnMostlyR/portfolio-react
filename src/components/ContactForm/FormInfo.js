import React from 'react';
import {FormattedMessage} from 'react-intl';
import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';

const Wrapper = styled.div`
  padding-bottom: 1rem;
`;

const FormInfoItemsList = styled.ul`
  list-style: none;
`;

const FormInfo = () => (
    <Wrapper>
      <FormInfoItemsList>
        <li>
          <FontAwesome name="info-circle"/> <span><FormattedMessage
            id={'portfolio.page.contact.requirement.one'}
            defaultMessage={'Please fill in all fields'}/></span>
        </li>
      </FormInfoItemsList>
    </Wrapper>
);

export default FormInfo;