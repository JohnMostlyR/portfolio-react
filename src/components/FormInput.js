import React, {Component} from 'react';
import PropTypes from 'prop-types';

class FormInput extends Component {
  static propTypes = {
    helperText: PropTypes.string,
    inputType: PropTypes.string,
    label: PropTypes.string.isRequired,
    maxLength: PropTypes.number,
    minLength: PropTypes.number,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    isTextArea: PropTypes.bool,
    validate: PropTypes.func,
    value: PropTypes.string,
  };

  static defaultProps = {
    inputType: 'text',
    isTextArea: false,
  };

  state = {
    value: this.props.value,
    error: false,
    _hasFocus: false,
    _height: '',
  };

  componentWillReceiveProps(update) {
    this.setState({value: update.value});
  }

  onChange = (ev) => {
    const value = ev.target.value;
    let _height = '';

    if (this.refs.textAreaElement) {
      if (this.refs.textAreaElement.scrollHeight > this.refs.textAreaElement.clientHeight) {
        _height = this.refs.textAreaElement.scrollHeight;
      } else {
        _height = this.refs.textAreaElement.clientHeight;
      }
    }

    this.setState({value, _height});
  };

  onBlur = (ev) => {
    const name = this.props.name;
    const value = ev.target.value;
    const error = this.props.validate ? this.props.validate(value) : false;
    const _hasFocus = false;

    this.setState({value, error, _hasFocus});

    this.props.onChange({name, value, error});
  };

  onFocus = () => {
    this.setState({_hasFocus: true,});
  };

  setFocus = (ev) => {
    if (this.refs.inputElement) {
      this.refs.inputElement.focus();
    } else if (this.refs.textAreaElement) {
      this.refs.textAreaElement.focus();
    }
  };

  render() {
    const renderInput = () => (
      <input
        className="c-portfolio-form__input"
        type={this.props.inputType}
        name={this.props.name}
        onChange={this.onChange}
        onFocus={this.onFocus}
        onBlur={this.onBlur}
        value={this.state.value}
        ref='inputElement'
      />
    );

    const renderTextArea = () => (
      <textarea
        style={{height: this.state._height}}
        className="c-portfolio-form__input c-portfolio-form__input--textarea"
        name={this.props.name}
        onBlur={this.onBlur}
        onChange={this.onChange}
        onFocus={this.onFocus}
        value={this.state.value}
        ref='textAreaElement'
      />
    );

    const renderHelperText = () => {
      if (!this.props.helperText) return '';

      let helperText = this.props.helperText;

      if (this.props.minLength || this.props.maxLength) {
        helperText = `${helperText} Should be between ${this.props.minLength} and ${this.props.maxLength} characters. Is now: ${this.state.value.length}`;
      }

      if (this.state._hasFocus || this.state.error || !this.state.value) {
        return helperText;
      } else if (!this.state.error) {
        return 'Thanks!';
      }

      return '';
    };

    return (
      <div className="c-portfolio-form__input-group">
        <label
          className="c-portfolio-form__label"
          htmlFor="name"
          onClick={this.setFocus}
        >
          <span
            className={`c-portfolio-form__label-content${(this.state._hasFocus || this.state.value) ? ' c-portfolio-form__label-content--on-focus' : ''}`}>{this.props.label}<span
            className="c-portfolio-form__label-content--placeholder"><i>{this.props.placeholder}</i></span></span>
        </label>
        {(this.props.isTextArea) ? renderTextArea() : renderInput()}
        <div className="c-portfolio-form__feedback-group">
          <span
            className={`c-portfolio-form__error-message${(this.state._hasFocus || !this.state.error) ? '' : ' is-active'}`}>{`${this.state.error} `}</span>
          <span
            className={'c-portfolio-form__helper-text is-active'}>{renderHelperText()}</span>
        </div>
      </div>
    );
  }
}

export default FormInput;
