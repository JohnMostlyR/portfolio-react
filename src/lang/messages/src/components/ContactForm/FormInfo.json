[
  {
    "id": "portfolio.page.contact.requirement.one",
    "defaultMessage": "Please fill in all fields"
  }
]