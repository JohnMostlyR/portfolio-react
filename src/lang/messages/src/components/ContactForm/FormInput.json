[
  {
    "id": "portfolio.page.contact.form_input.helper_text.range",
    "defaultMessage": "Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}"
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.valid",
    "defaultMessage": "Thanks!"
  }
]