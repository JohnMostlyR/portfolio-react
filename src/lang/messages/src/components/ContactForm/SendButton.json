[
  {
    "id": "send_button.idle",
    "description": "Send button value in normal, idle, state.",
    "defaultMessage": "Send"
  },
  {
    "id": "send_button.sending",
    "description": "Send button value when data is send.",
    "defaultMessage": "Sending..."
  },
  {
    "id": "send_button.success",
    "description": "Send button value when data is successfully sent.",
    "defaultMessage": "Sent! I'll respond as soon as possible."
  },
  {
    "id": "send_button.error",
    "description": "Send button value when an error occurred.",
    "defaultMessage": "Failed. Please try again"
  }
]