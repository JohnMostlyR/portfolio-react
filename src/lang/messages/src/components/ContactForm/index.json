[
  {
    "id": "portfolio.page.contact.form_input.helper_text.subject.text",
    "defaultMessage": "Please provide a subject for your message."
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.subject.label",
    "defaultMessage": "Subject"
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.subject.error",
    "defaultMessage": "A subject is required."
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.your_message.text",
    "defaultMessage": "Please add your message here."
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.your_message.label",
    "defaultMessage": "Your message"
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.your_message.error",
    "defaultMessage": "A message is required."
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.name.text",
    "defaultMessage": "Please add your name here."
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.name.label",
    "defaultMessage": "Your name"
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.name.placeholder",
    "defaultMessage": "(e.g. John Doo)"
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.name.error",
    "defaultMessage": "Your name is required."
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.email.text",
    "defaultMessage": "Please add your email address here."
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.email.label",
    "defaultMessage": "Your email address"
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.email.placeholder",
    "defaultMessage": "(e.g. my@email.com)"
  },
  {
    "id": "portfolio.page.contact.form_input.helper_text.email.error",
    "defaultMessage": "Your email address is required."
  }
]