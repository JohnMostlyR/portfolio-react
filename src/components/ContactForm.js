import React, {Component} from 'react';
import isEmail from 'validator/lib/isEmail';
import isLength from 'validator/lib/isLength';
import FormInput from './FormInput';

class ContactForm extends Component {
  state = {
    field: {
      subject: '',
      message: '',
      name: '',
      email: '',
    },
    fieldError: {},
    _sendStatus: 'IDLE', // IDLE, SENDING, SUCCESS, ERROR
  };

  validateForm = () => {
    const field = this.state.field;
    const fieldError = this.state.fieldError;
    const errMessages = Object.keys(fieldError).filter((key) => fieldError[key]);

    if (!field.email) {
      return true;
    }

    if (!field.message) {
      return true;
    }

    if (!field.name) {
      return true;
    }

    if (!field.subject) {
      return true;
    }

    if (errMessages.length) {
      return true;
    }

    return false;
  };

  onInputChange = ({name, value, error}) => {
    const field = this.state.field;
    const fieldError = this.state.fieldError;

    field[name] = value;
    fieldError[name] = error;

    this.setState({field, fieldError, _sendStatus: 'IDLE'});
  };

  _checkResponseStatus = (response) => {
    if (response.ok) {
      return Promise.resolve(response);
    } else {
      return Promise.reject(new Error(response.statusText));
    }
  };

  _parseJSON = (response) => {
    response.json();
  };

  onFormSubmit = (ev) => {
    ev.preventDefault();

    if (this.validateForm()) {
      return;
    }

    this.setState({_sendStatus: 'SENDING'});
    const field = this.state.field;

    fetch('https://meester-johan.info/contact-request', {
      method: 'post',
      headers: new Headers(
        {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        }
      ),
      body: JSON.stringify({
        email: field.email,
        message: field.message,
        name: field.name,
        subject: field.subject
      }),
    })
      .then(res => res.json())
      .then(data => {
        if (data.success) {
          this.setState({_sendStatus: 'SUCCESS'});
          console.log('Request succeeded with JSON response', data);
        } else {
          this.setState({_sendStatus: 'ERROR'});
          console.log('Request failed with JSON response', data);
          this.validateForm();
        }
      })
      .catch(error => {
        this.setState({_sendStatus: 'ERROR'});
        console.log(`Request failed with error: ${error}`);
      });
  };

  render() {
    const renderSendButton = () => {
      return {
        IDLE: <button
          className="c-portfolio-form__button"
          disabled={this.validateForm()}>
          <span><span
            className="fa fa-paper-plane c-portfolio-form__button-icon"
            aria-hidden="true"></span> Send</span>
        </button>,
        SENDING: <button
          className="c-portfolio-form__button"
          disabled>
          <span><span
            className="fa fa-paper-plane c-portfolio-form__button-icon"
            aria-hidden="true"></span> Sending...</span>
        </button>,
        SUCCESS: <button
          className="c-portfolio-form__button"
          disabled>
          <span><span
            className="fa fa-check c-portfolio-form__button-icon"
            aria-hidden="true"></span> Sent!</span>
        </button>,
        ERROR: <button
          className="c-portfolio-form__button"
          disabled={this.validateForm()}>
          <span><span
            className="fa fa-frown-o c-portfolio-form__button-icon"
            aria-hidden="true"></span> Failed. Please try again</span>
        </button>,
      }[this.state._sendStatus];
    };

    return (
      <form id="js-portfolio-form" className="c-portfolio-form" name="contact-form" action="#"
            acceptCharset="utf-8" method="post" onSubmit={this.onFormSubmit}>
        <div className="c-portfolio-form__frame">
          <div className="c-portfolio-form__content">
            <div className="c-portfolio-form__info">
              <ul className="c-portfolio-form__info-list">
                <li><span className="fa fa-info-circle" aria-hidden="true"/> Please fill in all fields
                </li>
              </ul>
            </div>
            <div className="c-portfolio-form__body">
              <FormInput
                helperText="Please provide a subject for your message."
                label="Subject"
                maxLength={50}
                minLength={3}
                name="subject"
                onChange={this.onInputChange}
                validate={val => (isLength(val, {min: 3, max: 50})) ? false : 'A subject is required.'}
                value={this.state.field.subject}
              />
              <FormInput
                helperText="Please add your message here."
                label="Your message"
                maxLength={300}
                minLength={5}
                name="message"
                onChange={this.onInputChange}
                isTextArea={true}
                validate={val => (isLength(val, {min: 5, max: 300})) ? false : 'A message is required.'}
                value={this.state.field.message}
              />
              <FormInput
                helperText="Please add your name here."
                label="Your name"
                maxLength={50}
                minLength={2}
                name="name"
                placeholder="(e.g. John Doo)"
                onChange={this.onInputChange}
                validate={val => (isLength(val, {min: 2, max: 50})) ? false : 'Your name is required.'}
                value={this.state.field.name}
              />
              <FormInput
                helperText="Please add your email address here."
                inputType="email"
                label="Your email address"
                name="email"
                placeholder="(e.g. my@email.com)"
                onChange={this.onInputChange}
                validate={val => (isEmail(val)) ? false : 'Your email address is required.'}
                value={this.state.field.email}
              />
            </div>
            <div className="c-portfolio-form__footer">
              {
                renderSendButton()
              }
            </div>
          </div>
        </div>
      </form>
    );
  }
}

export default ContactForm;
