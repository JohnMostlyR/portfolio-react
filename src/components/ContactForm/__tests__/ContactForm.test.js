import React from 'react';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

import ContactForm from '../index';
import FormInput from '../FormInput';

describe('ContactForm', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = mount(<ContactForm/>);
    instance = wrapper.childAt(0).instance();
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render and match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('TODO: should validate the user\'s input', () => {
    // const spy = jest.spyOn(instance, 'onInputChange');
    // const input = wrapper.find(FormInput).first().find('input');
    //
    // instance.onChange({name: 'subject', value: 'Form field input', error: false});
    // expect(spy).toHaveBeenCalled();
    // console.log(instance.state);
  });

  describe('TODO: user input is not valid', () => {
    it('should give the user feedback on the mistakes');
    it('should not allow the user to send the form');
  });

  describe('TODO: user input is valid', () => {
    it('should give the user feedback that the input is valid');
    it('should enable the send button');
  });

  describe('TODO: the user clicks the `send` button', () => {
    it('should not do anything as long as the input is not valid');
    it('should send the filled in information when the input is valid');
  })
});