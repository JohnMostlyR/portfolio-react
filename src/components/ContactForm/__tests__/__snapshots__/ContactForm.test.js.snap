// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should render and match the snapshot 1`] = `
<Component>
  <ContactForm
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <styled.form
      onSubmit={[Function]}
    >
      <form
        className="sc-VigVT aeRQa"
        onSubmit={[Function]}
      >
        <div>
          <div>
            <FormInfo>
              <styled.div>
                <div
                  className="sc-bdVaJa dvWVfn"
                >
                  <styled.ul>
                    <ul
                      className="sc-bwzfXH eHurHK"
                    >
                      <li>
                        <FontAwesome
                          name="info-circle"
                        >
                          <span
                            aria-hidden={true}
                            className="fa fa-info-circle"
                          />
                        </FontAwesome>
                         
                        <span>
                          <FormattedMessage
                            defaultMessage="Please fill in all fields"
                            id="portfolio.page.contact.requirement.one"
                          >
                            <IntlProvider
                              defaultLocale="en"
                              locale="en"
                            >
                              <FormattedMessage
                                defaultMessage="Please fill in all fields"
                                id="portfolio.page.contact.requirement.one"
                                values={Object {}}
                              >
                                <span>
                                  Please fill in all fields
                                </span>
                              </FormattedMessage>
                            </IntlProvider>
                          </FormattedMessage>
                        </span>
                      </li>
                    </ul>
                  </styled.ul>
                </div>
              </styled.div>
            </FormInfo>
            <div>
              <FormInput
                helperText="Please provide a subject for your message."
                inputType="text"
                isTextArea={false}
                label="Subject"
                maxLength={50}
                minLength={3}
                name="subject"
                onChange={[Function]}
                validate={[Function]}
                value=""
              >
                <styled.div>
                  <div
                    className="sc-htpNat cdCmwn"
                  >
                    <styled.label
                      htmlFor="name"
                      onClick={[Function]}
                    >
                      <label
                        className="sc-bZQynM fyzOqO"
                        htmlFor="name"
                        onClick={[Function]}
                      >
                        <styled.span
                          hasFocus=""
                        >
                          <span
                            className="sc-htoDjs iyMcMx"
                          >
                            Subject
                            <styled.span
                              hasFocus=""
                            >
                              <span
                                className="sc-gzVnrw jDpwPe"
                              >
                                <i />
                              </span>
                            </styled.span>
                          </span>
                        </styled.span>
                      </label>
                    </styled.label>
                    <styled.input
                      innerRef={[Function]}
                      name="subject"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="text"
                      value=""
                    >
                      <input
                        className="sc-bxivhb jqrJDT"
                        name="subject"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="text"
                        value=""
                      />
                    </styled.input>
                    <div>
                      <styled.span
                        showError={false}
                      >
                        <span
                          className="sc-dnqmqq cYQxgH"
                        >
                          false 
                        </span>
                      </styled.span>
                      <styled.span>
                        <span
                          className="sc-iwsKbI bRezmp"
                        >
                          <span>
                            Please provide a subject for your message.
                            Â 
                            <FormattedMessage
                              defaultMessage="Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}"
                              id="portfolio.page.contact.form_input.helper_text.range"
                              values={
                                Object {
                                  "count": 0,
                                  "maxLength": 50,
                                  "minLength": 3,
                                }
                              }
                            >
                              <IntlProvider
                                defaultLocale="en"
                                locale="en"
                              >
                                <FormattedMessage
                                  defaultMessage="Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}"
                                  id="portfolio.page.contact.form_input.helper_text.range"
                                  values={
                                    Object {
                                      "count": 0,
                                      "maxLength": 50,
                                      "minLength": 3,
                                    }
                                  }
                                >
                                  <span>
                                    Should be between 3 and 50 characters. Is now: 0
                                  </span>
                                </FormattedMessage>
                              </IntlProvider>
                            </FormattedMessage>
                          </span>
                        </span>
                      </styled.span>
                    </div>
                  </div>
                </styled.div>
              </FormInput>
              <FormInput
                helperText="Please add your message here."
                inputType="text"
                isTextArea={true}
                label="Your message"
                maxLength={300}
                minLength={5}
                name="message"
                onChange={[Function]}
                validate={[Function]}
                value=""
              >
                <styled.div>
                  <div
                    className="sc-htpNat cdCmwn"
                  >
                    <styled.label
                      htmlFor="name"
                      onClick={[Function]}
                    >
                      <label
                        className="sc-bZQynM fyzOqO"
                        htmlFor="name"
                        onClick={[Function]}
                      >
                        <styled.span
                          hasFocus=""
                        >
                          <span
                            className="sc-htoDjs iyMcMx"
                          >
                            Your message
                            <styled.span
                              hasFocus=""
                            >
                              <span
                                className="sc-gzVnrw jDpwPe"
                              >
                                <i />
                              </span>
                            </styled.span>
                          </span>
                        </styled.span>
                      </label>
                    </styled.label>
                    <styled.textarea
                      innerRef={[Function]}
                      name="message"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      style={
                        Object {
                          "height": "",
                        }
                      }
                      value=""
                    >
                      <textarea
                        className="sc-EHOje dBDhYl"
                        name="message"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        style={
                          Object {
                            "height": "",
                          }
                        }
                        value=""
                      />
                    </styled.textarea>
                    <div>
                      <styled.span
                        showError={false}
                      >
                        <span
                          className="sc-dnqmqq cYQxgH"
                        >
                          false 
                        </span>
                      </styled.span>
                      <styled.span>
                        <span
                          className="sc-iwsKbI bRezmp"
                        >
                          <span>
                            Please add your message here.
                            Â 
                            <FormattedMessage
                              defaultMessage="Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}"
                              id="portfolio.page.contact.form_input.helper_text.range"
                              values={
                                Object {
                                  "count": 0,
                                  "maxLength": 300,
                                  "minLength": 5,
                                }
                              }
                            >
                              <IntlProvider
                                defaultLocale="en"
                                locale="en"
                              >
                                <FormattedMessage
                                  defaultMessage="Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}"
                                  id="portfolio.page.contact.form_input.helper_text.range"
                                  values={
                                    Object {
                                      "count": 0,
                                      "maxLength": 300,
                                      "minLength": 5,
                                    }
                                  }
                                >
                                  <span>
                                    Should be between 5 and 300 characters. Is now: 0
                                  </span>
                                </FormattedMessage>
                              </IntlProvider>
                            </FormattedMessage>
                          </span>
                        </span>
                      </styled.span>
                    </div>
                  </div>
                </styled.div>
              </FormInput>
              <FormInput
                helperText="Please add your name here."
                inputType="text"
                isTextArea={false}
                label="Your name"
                maxLength={50}
                minLength={2}
                name="name"
                onChange={[Function]}
                placeholder="(e.g. John Doo)"
                validate={[Function]}
                value=""
              >
                <styled.div>
                  <div
                    className="sc-htpNat cdCmwn"
                  >
                    <styled.label
                      htmlFor="name"
                      onClick={[Function]}
                    >
                      <label
                        className="sc-bZQynM fyzOqO"
                        htmlFor="name"
                        onClick={[Function]}
                      >
                        <styled.span
                          hasFocus=""
                        >
                          <span
                            className="sc-htoDjs iyMcMx"
                          >
                            Your name
                            <styled.span
                              hasFocus=""
                            >
                              <span
                                className="sc-gzVnrw jDpwPe"
                              >
                                <i>
                                  (e.g. John Doo)
                                </i>
                              </span>
                            </styled.span>
                          </span>
                        </styled.span>
                      </label>
                    </styled.label>
                    <styled.input
                      innerRef={[Function]}
                      name="name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="text"
                      value=""
                    >
                      <input
                        className="sc-bxivhb jqrJDT"
                        name="name"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="text"
                        value=""
                      />
                    </styled.input>
                    <div>
                      <styled.span
                        showError={false}
                      >
                        <span
                          className="sc-dnqmqq cYQxgH"
                        >
                          false 
                        </span>
                      </styled.span>
                      <styled.span>
                        <span
                          className="sc-iwsKbI bRezmp"
                        >
                          <span>
                            Please add your name here.
                            Â 
                            <FormattedMessage
                              defaultMessage="Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}"
                              id="portfolio.page.contact.form_input.helper_text.range"
                              values={
                                Object {
                                  "count": 0,
                                  "maxLength": 50,
                                  "minLength": 2,
                                }
                              }
                            >
                              <IntlProvider
                                defaultLocale="en"
                                locale="en"
                              >
                                <FormattedMessage
                                  defaultMessage="Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}"
                                  id="portfolio.page.contact.form_input.helper_text.range"
                                  values={
                                    Object {
                                      "count": 0,
                                      "maxLength": 50,
                                      "minLength": 2,
                                    }
                                  }
                                >
                                  <span>
                                    Should be between 2 and 50 characters. Is now: 0
                                  </span>
                                </FormattedMessage>
                              </IntlProvider>
                            </FormattedMessage>
                          </span>
                        </span>
                      </styled.span>
                    </div>
                  </div>
                </styled.div>
              </FormInput>
              <FormInput
                helperText="Please add your email address here."
                inputType="email"
                isTextArea={false}
                label="Your email address"
                name="email"
                onChange={[Function]}
                placeholder="(e.g. my@email.com)"
                validate={[Function]}
                value=""
              >
                <styled.div>
                  <div
                    className="sc-htpNat cdCmwn"
                  >
                    <styled.label
                      htmlFor="name"
                      onClick={[Function]}
                    >
                      <label
                        className="sc-bZQynM fyzOqO"
                        htmlFor="name"
                        onClick={[Function]}
                      >
                        <styled.span
                          hasFocus=""
                        >
                          <span
                            className="sc-htoDjs iyMcMx"
                          >
                            Your email address
                            <styled.span
                              hasFocus=""
                            >
                              <span
                                className="sc-gzVnrw jDpwPe"
                              >
                                <i>
                                  (e.g. my@email.com)
                                </i>
                              </span>
                            </styled.span>
                          </span>
                        </styled.span>
                      </label>
                    </styled.label>
                    <styled.input
                      innerRef={[Function]}
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="email"
                      value=""
                    >
                      <input
                        className="sc-bxivhb jqrJDT"
                        name="email"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        type="email"
                        value=""
                      />
                    </styled.input>
                    <div>
                      <styled.span
                        showError={false}
                      >
                        <span
                          className="sc-dnqmqq cYQxgH"
                        >
                          false 
                        </span>
                      </styled.span>
                      <styled.span>
                        <span
                          className="sc-iwsKbI bRezmp"
                        >
                          <span>
                            Please add your email address here.
                            Â 
                          </span>
                        </span>
                      </styled.span>
                    </div>
                  </div>
                </styled.div>
              </FormInput>
            </div>
            <styled.div>
              <div
                className="sc-jTzLTM cpVoso"
              >
                <SendButton
                  buttonState="idle"
                  disabled={true}
                >
                  <styled.button
                    disabled={true}
                  >
                    <button
                      className="sc-gZMcBi feqOeo"
                      disabled={true}
                    >
                      <span>
                        <Styled(FontAwesome)
                          name="paper-plane"
                        >
                          <FontAwesome
                            className="sc-gqjmRU fOMCdq"
                            name="paper-plane"
                          >
                            <span
                              aria-hidden={true}
                              className="fa fa-paper-plane sc-gqjmRU fOMCdq"
                            />
                          </FontAwesome>
                        </Styled(FontAwesome)>
                        <span>
                          <FormattedMessage
                            defaultMessage="Send"
                            id="send_button.idle"
                          >
                            <IntlProvider
                              defaultLocale="en"
                              locale="en"
                            >
                              <FormattedMessage
                                defaultMessage="Send"
                                id="send_button.idle"
                                values={Object {}}
                              >
                                <span>
                                  Send
                                </span>
                              </FormattedMessage>
                            </IntlProvider>
                          </FormattedMessage>
                        </span>
                      </span>
                    </button>
                  </styled.button>
                </SendButton>
              </div>
            </styled.div>
          </div>
        </div>
      </form>
    </styled.form>
  </ContactForm>
</Component>
`;
